### DISCUSSION TITLE
Respect `DISALLOW_FILE_MODS`, but why?

### DISCUSSION THREAD

**Discussion by vreoo** [rhubarbgroup/.github#38] [Support (ROC)] [answered]
Hello there,

In your recent 2.5.0 update, we noticed that you added this change: [`df87cc2`](https://github.com/rhubarbgroup/redis-cache/commit/df87cc2ddadeee09b28c4b6b7d5c3e1164ae7acd)
```php
/**
     * Determines whether object cache file modifications are allowed.
     *
     * @return bool
     */
    function is_file_mod_allowed() {
        return apply_filters(
            'file_mod_allowed',
            ! defined( 'DISALLOW_FILE_MODS' ) || ! DISALLOW_FILE_MODS,
            'object_cache_dropin'
        );
    }
``` 
And this:
```php
 if ( ! $this->is_file_mod_allowed() ) {
            return new WP_Error( 'disallowed', __( 'File modifications are not allowed.', 'redis-cache' ) );
        }
```

But I don't fully get it, what is the reason behind this change?

We have `DISALLOW_FILE_MODS` set to true on our environment so when we updated your plugin it started to give us a not writable error so we had to manually ignore `DISALLOW_FILE_MODS` for the redis object cache by adding a function to `functions.php` so we can make the plugin work again.

What is the best practice here in our case in your opinion? how will that affect the environment?

Thanks,

**Comment by tillkruss** [comment, marked as answer]
Those are essentially checks for [`wp_is_file_mod_allowed()`](https://developer.wordpress.org/reference/functions/wp_is_file_mod_allowed/).

If the filesystem is not writeable (`DISALLOW_FILE_MODS = true`) then we shouldn't try to write the drop-in on the filesystem.

In your environment, I'd suggest putting the `object-cache.php` in place by hand, or using WP CLI.
